### PR TITLE
Zachmu/insert update delete perf

### PR DESCRIPTION
**Pull request by zachmu**

### CODE REVIEW

File: sql/core.go [hunk 1]
```diff
@@ -205,28 +205,76 @@ type IndexableTable interface {
...
+	// updated, followed by a call to Close() when all rows have been processed.
+	Updater(ctx *Context) RowUpdater
 }
 
-// Updater allows rows to be updated.
-type Updater interface {
+type RowUpdater interface {
 	// Update the given row. Provides both the old and new rows.
 	Update(ctx *Context, old Row, new Row) error
+	// Close finalizes the delete operation, persisting the result.
```
Comment (Hydrocharged) [inline sql/core.go]: I think you forgot to update the comment here, should be update.

File: sql/core.go [hunk 2]
```diff
@@ -205,28 +205,76 @@ type IndexableTable interface {
...
+type RowDeleter interface {
+	// Delete deletes the given row. Returns ErrDeleteRowNotFound if the row was not found. Delete will be called once for
+	// each row to process for the delete operation, which may involve many rows. After all rows have been processed,
+	// Close is called.
+	Delete(*Context, Row) error
+	// Close finalizes the delete operation, persisting the result.
+	Close(*Context) error
+}
+
+type Closer interface {
```
Comment (Hydrocharged) [inline sql/core.go]: Since we've got this `Closer` interface, why not embed it in the `Row*` interfaces as well?

Comment (zachmu) [reply sql/core.go]: Good idea

File: sql/core.go (hunk 1 above)
Comment (zachmu) [reply sql/core.go]: Done

**Pull request merged**